### ISSUE TITLE
Adding training scripts

### ISSUE THREAD

**Issue by isamu-isozaki** [lucidrains/med-seg-diff-pytorch#2]
Hi! Thanks for the code! My friend and I are making a training script using [this dataset](https://challenge.isic-archive.com/data/)

**Comment by isamu-isozaki** [comment]
Will make a pr if you want!

**Comment by lucidrains** [comment]
sure, thank you!

**Comment by lucidrains** [comment]
@isamu-isozaki highly recommend the [huggingface accelerate library](https://huggingface.co/docs/accelerate/index)

**Comment by isamu-isozaki** [comment]
@lucidrains Sounds good! Quite a fan of it already. We are mostly done with the script but will integrate accelerate and open a pr

**Comment by lucidrains** [comment]
:pray: :rocket:

**Comment by Time9Y** [comment]
> Hi! Thanks for the code! My friend and I are making a training script using [this dataset](https://challenge.isic-archive.com/data/)

Thank you very much for your contribution and look forward to your code!

**Comment by Time9Y** [comment]
Thanks for the code, I tried to add the training code but the result was not satisfactory.

**Comment by isamu-isozaki** [comment]
@Time9Y yup my friend and I are having trouble getting good results too. It kinda works on 64x64 resolution but on 128 it just breaks.
I'll try checking the paper for their params

**Comment by lucidrains** [comment]
oh this is disappointing

could i have made an error?

they released a public official repository btw

**Comment by isamu-isozaki** [comment]
@lucidrains Haha no most likely we screwed up a bit. And sounds good! Will check that out. I'll add our script as a pr in a bit. Just cleaning up since it's a bit messy but maybe we can get your opinion too!

**Comment by Time9Y** [comment]
@lucidrains This should be my problem, the code you provided is clean and efficient.

**Comment by Time9Y** [comment]
@isamu-isozaki I've seen the code you submitted and it's so cool. Hope I can use it soon!

**Comment by isamu-isozaki** [comment]
@Time9Y Thank you! Will let you know when finished

**Comment by Time9Y** [comment]
@isamu-isozaki I've received the code, it's awesome! During training process, how many iterations did you get good results after.

**Comment by Time9Y** [comment]
@isamu-isozaki Amazing, I achieved great results in 50 iterations, thanks for your contribution! This code will help our research progress.

**Comment by isamu-isozaki** [comment]
@Time9Y Oh awesome! Glad it helped. What parameters did you guys use out of curiosity?

**Comment by Time9Y** [comment]
@isamu-isozaki The parameters are basically similar to yours, while referring to this code. https://github.com/WuJunde/MedSegDiff
lr 1e-4 eopch 150 batch_size 4 UNet (1, 2, 3, 4) not use ddim

**Comment by isamu-isozaki** [comment]
@Time9Y Thanks! My friend will appreciate that a lot

**Comment by lucidrains** [comment]
> @isamu-isozaki Amazing, I achieved great results in 50 iterations, thanks for your contribution! This code will help our research progress.

great to hear! :pray:

@isamu-isozaki  thank you so much for taking the time to contribute the training script!

**Comment by isamu-isozaki** [comment]
@lucidrains Np! I wanted to contribute later to your muse repo so this was a bit of a practice run. And also, @dsbuddy helped a lot.

**Comment by lucidrains** [comment]
@dsbuddy thank you Daniel :100:  !

**Comment by Luoqian524** [comment]
> 呃！我想稍后为您的 muse 存储库做出贡献，因此这有点像一次练习。而且，帮助很大。

Is it okay if I want your code? I would appreciate if you share your code with me.
